src/main/scala/cromwell/binding/Call.scala

```diff
@@ -1,11 +1,56 @@
 package cromwell.binding
 
 import cromwell.binding.types.WdlType
-import cromwell.binding.values.WdlValue
 import cromwell.parser.AstTools
+import cromwell.parser.WdlParser.{SyntaxError, Terminal, Ast}
+import cromwell.parser.AstTools.EnhancedAstNode
 
 import scala.util.{Success, Try}
 
+object Call {
+  def apply(ast: Ast,
+            namespaces: Seq[WdlNamespace],
+            tasks: Seq[Task],
+            wdlSyntaxErrorFormatter: WdlSyntaxErrorFormatter): Call = {
+    val alias: Option[String] = ast.getAttribute("alias") match {
+      case x: Terminal => Option(x.getSourceString)
+      case _ => None
+    }
+
+    val taskName = ast.getAttribute("task").sourceString()
+    val task = WdlNamespace.findTask(taskName, namespaces, tasks) getOrElse {
+      throw new SyntaxError(wdlSyntaxErrorFormatter.callReferencesBadTaskName(ast, taskName))
+    }
+
+    val inputMappings = processCallInput(ast)
+
+    inputMappings foreach { inputKv =>
+      task.inputs find { taskInput => taskInput._1 == inputKv._1 } getOrElse {
+        /*
+          FIXME-ish
+          It took me a while to figure out why this next part is necessary and it's kind of hokey.
+          All the syntax highlighting requires ASTs and this is a way to get the input's AST back.
+          Perhaps an intermediary object that contains the AST as well and then the normal Map in the Call?
+         */
```

Comment:
I think at least one. I think that's out of the purview of this PR though but now that both of us have marked this with a `FIXME` it should be gone through soonish